plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -233,9 +233,9 @@ public int compare(QuotaBalanceVO o1, QuotaBalanceVO o2) {
             // order is desc last item is the start item
             QuotaBalanceVO startItem = quotaBalance.get(quotaBalance.size() - 1);
             QuotaBalanceVO endItem = quotaBalance.get(0);
-            resp.setStartDate(startItem.getUpdatedOn());
+            resp.setStartDate(startDate);
```

Comment:
instead of startDate, pass a new object like new Date(startDate.getTime()) for the ref issue we discovered with the earlier PR. Alternatively, you may fix the setters and getters to do return or set using new Date() syntax